packages/avs-bot/src/main.ts

```diff
@@ -0,0 +1,335 @@
+    const config = {
+      store: this.storeEngine,
+      urls: APIClient.BACKEND.STAGING,
+    };
+    const apiClient = new APIClient(config);
+    const account = new Account(apiClient, async (): Promise<CRUDEngine> => this.storeEngine);
+    this.account = account;
+
+    apiClient.on(APIClient.TOPIC.ACCESS_TOKEN_REFRESH, async (accessToken: AccessTokenData) => {
+      await this.storeEngine.updateOrCreate(AUTH_TABLE_NAME, AUTH_ACCESS_TOKEN_KEY, accessToken);
+    });
+
+    apiClient.on(APIClient.TOPIC.COOKIE_REFRESH, async (cookie?: Cookie) => {
+      console.log('new cookie', cookie);
+      if (!cookie) {
+        return;
+      }
+      this.cookie = cookie;
+      const entity = {expiration: cookie.expiration, zuid: cookie.zuid};
+      await this.storeEngine.updateOrCreate(AUTH_TABLE_NAME, AUTH_COOKIE_KEY, entity);
+    });
+
+    await this.getCookie();
+
+    let userId: string;
+    try {
+      if (!this.cookie) {
+        throw new Error('No cookie found');
+      }
+      userId = (await account.init(CLIENT_TYPE, this.cookie)).userId;
+    } catch (error) {
+      console.log('Failed to get new access token with cookie');
+      userId = (await account.login(loginData)).userId;
+    }
+    const {
+      localClient: {id: clientId},
+    } = await account.initClient(loginData);
+    await this.initAvs(userId, clientId);
+
+    account.on(PayloadBundleType.CALL, call => {
+      this.onIncomingCallMessage(call);
+    });
+
+    await account.listen();
+  }
+
+  onIncomingCallMessage = (callMessage: CallMessage) => {
+    try {
+      const callContent: CallingContent = callMessage.content as CallingContent;
+      if (!callMessage.fromClientId) {
+        throw new Error('callMessage.fromClientId not found');
+      }
+      const avsResult = this.wCall!.recvMsg(
+        this.wUser!,
+        callContent,
+        callContent.length,
+        new Date().getSeconds(),
+        new Date(callMessage.timestamp).getSeconds(),
+        callMessage.conversation,
+        callMessage.from,
+        callMessage.fromClientId,
+      );
+      if (avsResult !== 0) {
+        throw new Error(`recvMsg failed with code "${avsResult}"`);
+      }
+      this.answerCall(callMessage);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  answerCall = (call: CallMessage) => {
+    if (!this.wCall || !this.wUser) {
+      return;
+    }
+    console.log('\n ANSWERING CALL \n');
+    this.wCall.answer(this.wUser, call.conversation, CALL_TYPE.NORMAL, 0);
+  };
+
+  async getCookie(): Promise<Cookie | undefined> {
+    try {
+      const {expiration, zuid} = await this.storeEngine.read(AUTH_TABLE_NAME, AUTH_COOKIE_KEY);
+      const cookie = new Cookie(zuid, expiration);
+      this.cookie = cookie;
+      return cookie;
+    } catch (error) {
+      console.log('no cookie found');
+      return undefined;
+    }
+  }
+
+  async initAvs(selfUserId: string, clientId: string): Promise<{wCall: Wcall; wUser: number}> {
```

Comment:
I think it would be good to separate the calling related code from the account related code in a separate file.

Reply:
https://github.com/wireapp/wire-web-packages/pull/3070/commits/61aa6120a1f8ef59ff36def1aea7e40d4d9aaccb